compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/values/JvmObjectGen.java

```diff
@@ -145,6 +154,22 @@ public void createAndSplitCallMethod(ClassWriter cw, List<BIRNode.BIRFunction> f
             } else {
                 jvmCastGen.addBoxInsn(mv, retType);
             }
+            if (func.name.value.equals("attach") &&
```

Comment:
should we check remoteManagement build option?

Reply:
We can add that as well after we merge https://github.com/ballerina-platform/ballerina-lang/pull/42336. The option will be available in the `CompilerContext`.

Reply:
As per the latest design changes, the build option will be added through a compiler plugin from the `wso2.controlplane` module. Those changes will be added in a separate PR.